packages/devtools_app/lib/src/memory/flutter/memory_screen.dart

```diff
@@ -106,67 +106,36 @@ class MemoryBodyState extends State<MemoryBody> {
     );
   }
 
-  static const String liveFeed = 'Live Feed';
-  String memorySource = liveFeed;
-
-  Widget createMenuItem(String name) {
-    final rowChildren = memorySource == name
-        ? [
-            Icon(Icons.check, size: 12),
-            const SizedBox(width: 10),
-            Text(name, key: MemoryScreen.memorySourcesKey),
-          ]
-        : [
-            const SizedBox(width: 22),
-            Text(name, key: MemoryScreen.memorySourcesKey),
-          ];
-
-    return PopupMenuItem<String>(
-      value: name,
-      child: Row(
-        crossAxisAlignment: CrossAxisAlignment.center,
-        children: rowChildren,
-      ),
-    );
-  }
-
-  Widget _selectMemoryFile() {
-    final List<String> files = controller.memoryLog.offlineFiles();
-
-    final List<PopupMenuItem<String>> items = [
-      createMenuItem(liveFeed),
-    ];
-
-    for (var index = 0; index < files.length; index++) {
-      items.add(createMenuItem(files[index]));
-    }
-
-    return PopupMenuButton<String>(
-      key: MemoryScreen.popupSourceMenuButtonKey,
-      onSelected: (value) {
+  Widget _memorySourceDropdown() {
+    final files = controller.memoryLog.offlineFiles();
+
+    // First item is 'Live Feed', then followed by memory log filenames.
+    files.insert(0, MemoryController.liveFeed);
+
+    final allMemorySources = files.map<DropdownMenuItem<String>>((
+      String value,
+    ) {
+      return DropdownMenuItem<String>(
+        key: MemoryScreen.memorySourcesMenuItem,
+        value: value,
+        child: Text(
+          value,
+          key: MemoryScreen.memorySourcesKey,
+        ),
+      );
+    }).toList();
+
+    return DropdownButton<String>(
```

Comment:
instead of calling set state to maintain the value of controller.memorySource in the UI, do
```
return ValueListenableBuilder(
  valueListenable: controller.memorySourceNotifier,
  builder: (context, memorySource, _) {
    return DropdownButton<String>(
      ...
      value: memorySource,
      ...
      onChanged: controller.selectMemorySource,
    );
  }
);
```
And in the controller
```
ValueListenable get memorySourceNotifier => _memorySourceNotifier;
final _memorySourceNotifier = ValueNotifier<String>([starting value here]);

void selectMemorySource(String name) {
  _memorySourceNotifier.value = name;
}
```
This way you can eliminate the calls to set state because the state is managed by the ValueListenableBuilder widget.